test/test_asynchronous_process_fork.py

```diff
@@ -358,8 +358,10 @@ def test_decorated_dead_process(self):
         async def test():
             return await critical_decorated()
 
-        with self.assertRaises(ProcessExpired):
+        with self.assertRaises(ProcessExpired) as exc_ctx:
             asyncio.run(test())
+        self.assertEqual(exc_ctx.exception.exitcode, 123)
+        self.assertIsInstance(exc_ctx.exception.pid, int)
```

Comment:
For the async + concurrent tests I choose to simply assert the pid exists as an `int` instead of mocking `common.launch_process` to try and get the actual `pid` and comparing them. If you would prefer that though I can update.

Reply:
It's OK I guess. We cannot predict the PID value in this case.